test/recipes/01-test_symbol_presence.t

```diff
@@ -23,54 +24,97 @@ use platform;
-plan skip_all => "This is unsupported on MSYS, MinGW or MSWin32"
-    if $^O eq 'msys' or $^O eq 'MSWin32' or config('target') =~ m|^mingw|;
-plan skip_all => "Only useful when building shared libraries"
-    if disabled("shared");
-
-my @libnames = ("crypto", "ssl");
-my $testcount = scalar @libnames;
-
-plan tests => $testcount * 2;
+plan skip_all => "This is unsupported on platforms that don't have 'nm'"
+    unless IPC::Cmd::can_run('nm');
 
 note
     "NOTE: developer test!  It's possible that it won't run on your\n",
     "platform, and that's perfectly fine.  This is mainly for developers\n",
     "on Unix to check that our shared libraries are consistent with the\n",
-    "ordinals (util/*.num in the source tree), something that should be\n",
-    "good enough a check for the other platforms as well.\n";
-
-foreach my $libname (@libnames) {
- SKIP:
-    {
-        my $shlibname = platform->sharedlib("lib$libname");
-        my $shlibpath = bldtop_file($shlibname);
-        *OSTDERR = *STDERR;
-        *OSTDOUT = *STDOUT;
-        open STDERR, ">", devnull();
-        open STDOUT, ">", devnull();
-        my @nm_lines = map { s|\R$||; $_ } `nm -DPg $shlibpath 2> /dev/null`;
-        close STDERR;
-        close STDOUT;
-        *STDERR = *OSTDERR;
-        *STDOUT = *OSTDOUT;
-        skip "Can't run 'nm -DPg $shlibpath' => $?...  ignoring", 2
-            unless $? == 0;
-
-        my $bldtop = bldtop_dir();
-        my @def_lines;
+    "ordinals (util/*.num in the source tree), and that our static libraries\n",
+    "don't shared symbols, something that should be good enough a check for\n",
+    "the other platforms as well.\n";
+
+my %stlibname;
+my %shlibname;
+my %stlibpath;
+my %shlibpath;
+my %defpath;
+foreach (qw(crypto ssl)) {
+    $stlibname{$_} = platform->staticlib("lib$_");
+    $stlibpath{$_} = bldtop_file($stlibname{$_});
+    $shlibname{$_} = platform->sharedlib("lib$_") unless disabled('shared');
+    $shlibpath{$_} = bldtop_file($shlibname{$_})  unless disabled('shared');
+}
+
+my $testcount
+    =  1                        # Check for static library symbols duplicates
+    ;
+$testcount
+    += (scalar keys %shlibpath) # Check for missing symbols in shared lib
+    unless disabled('shared');
+
+plan tests => $testcount;
+
+######################################################################
+# Collect symbols
+# [3 tests per library]
+
+my %stsymbols;                  # Static library symbols
+my %shsymbols;                  # Shared library symbols
+my %defsymbols;                 # Symbols taken from ordinals
+foreach (sort keys %stlibname) {
+    my $stlib_cmd = "nm -Pg $stlibpath{$_} 2> /dev/null";
+    my $shlib_cmd = "nm -DPg $shlibpath{$_} 2> /dev/null";
+    my @stlib_lines;
+    my @shlib_lines;
+    *OSTDERR = *STDERR;
+    *OSTDOUT = *STDOUT;
+    open STDERR, ">", devnull();
+    open STDOUT, ">", devnull();
+    @stlib_lines = map { s|\R$||; $_ } `$stlib_cmd`; $? = 1;
```

Comment:
I don't understand the trailing `$? = 1` here. Can you explain it?

Reply:
Yes I can!  It's a remnant from an incantation that never made it into this PR.  Thanks for spotting it, I'm now removing it (at every spot)